OpenRA.Game/Graphics/Viewport.cs

```diff
@@ -74,6 +81,23 @@ public float Zoom
 		public static int TicksSinceLastMove = 0;
 		public static int2 LastMousePos;
 
+		float ClosestTo(IEnumerable<float> collection, float target)
```

Comment:
This is a private method, so there is no need to abstract away the array (which just hurts perf).  Just use `float[] collection` directly.